check_mk/check_galera.sh

```diff
@@ -0,0 +1,164 @@
+#!/bin/bash
+PROGNAME=`basename $0`
+VERSION="Version 1.0,"
+AUTHOR="Guillaume Coré <g@fridim.org>"
+
+ST_OK=0
+ST_WR=1
+ST_CR=2
+ST_UK=3
+
+print_version() {
+  echo "$VERSION $AUTHOR"
+}
+
+print_help() {
+  print_version $PROGNAME $VERSION
+  echo ""
+  echo "$PROGNAME is a Nagios plugin to monitor Galera cluster status."
+  echo ""
+  echo "$PROGNAME -u USER -p PASSWORD [-H HOST] [-P PORT] [-w SIZE] [-c SIZE] [-f FLOAT] [-0]"
+  echo ""
+  echo "Options:"
+  echo "  u)"
+  echo "    MySQL user."
+  echo "  p)"
+  echo "    MySQL password."
+  echo "  H)"
+  echo "    MySQL host. Default is localhost."
+  echo "  P)"
+  echo "    MySQL port. Default is 3306."
+  echo "  w)"
+  echo "    Sets minimum number of nodes in the cluster when WARNING is raised. (default is same as critical)."
+  echo "  c)"
+  echo "    Sets minimum number of nodes in the cluster when CRITICAL is raised. (default is 2)."
+  echo "  f)"
+  echo "    Sets critical value of wsrep_flow_control_paused (default is 0.1)."
+  echo "  0)"
+  echo "    Rise CRITICAL if the node is not primary"
+  exit $ST_UK
+}
+
+# default values
+crit=1
+warn=2
+port='3306'
+mysqlhost='localhost'
+fcp=0.1
+
+while getopts “hvu:p:H:P:w:c:f:0” OPTION; do
+  case $OPTION in
+    h)
+      print_help
+      exit $ST_UK
+      ;;
+    v)
+      print_version $PROGNAME $VERSION
+      exit $ST_UK
+      ;;
+    u)
+      mysqluser=$OPTARG
+      ;;
+    p)
+      password=$OPTARG
+      ;;
+    H)
+      mysqlhost=$OPTARG
+      ;;
+    P)
+      port=$OPTARG
+      ;;
+    w)
+      warn=$OPTARG
+      ;;
+    c)
+      crit=$OPTARG
+      ;;
+    f)
+      fcp=$OPTARG
+      ;;
+    0)
+      primary='TRUE'
+      ;;
+    ?)
+      echo "Unknown argument: $1"
+      print_help
+      exit $ST_UK
+      ;;
+  esac
+done
+
+mysqluser=root
+password=mydbms
```

Comment:
YES, oops.  The mysql servers aren't accessible off network but it's still bad to have it in there.